src/irmin-pack/irmin_pack.mli

```diff
@@ -79,13 +79,16 @@ module Atomic_write (K : Irmin.Type.S) (V : Irmin.Hash.S) : sig
 end
 
 module Pack (K : Irmin.Hash.S) : sig
+  module Index : module type of Index.Make (K)
```

Comment:
you probably don't need to expose the Index module here.

Reply:
I believe you're right, because of the way Index works currently. i.e. if a duplicate finds an existing instance, they will cohabit happily. Passing the index down from the top-level was an attempt to only have to instantiate it once, but is perhaps not worth it if Index is already behaving nicely with multiple instances?